ui-library-add-call-recording/api/transfer/BlobStorage.cs

```diff
@@ -37,6 +38,17 @@ public async Task<bool> UploadFileAsync(string blobName, Stream inStream)
                 await blobClient.UploadAsync(inStream);
                 return true;
             }
+
+            public string[] GetBlobNames(string prefix)
+            {
+                var blobs = this.containerClient.GetBlobs(Azure.Storage.Blobs.Models.BlobTraits.None, Azure.Storage.Blobs.Models.BlobStates.None, prefix);
+                var names = new List<string>();
+                foreach (var blob in blobs)
+                {
+                    names.Add(blob.Name);
+                }
+                return names.ToArray();
```

Comment:
this is hilarious you have to get an IEnumerator, push to a list, then convert to array

(I tried to find a nicer way and couldn't find a good one, if you return an IEnumerable<string> instead of a string[] you can do
```
var blobs = this.containerClient.GetBlobs(Azure.Storage.Blobs.Models.BlobTraits.None, Azure.Storage.Blobs.Models.BlobStates.None, prefix);
foreach (var blob in blobs) {
   yield return blob.Name
}
```
but not if you want to return an array)

Reply:
Yeah, I couldn't find a way to collect into an array directly. Dunno what's the equivalent of a `Vector` in C#.

This is definitely my unfamiliarity with C# showing up.